### PR TITLE
Change license to GPLv3+

### LINKED ISSUE #198: License of the package - consider move to GPL-3+
In a thread in the emacs-devel maillist, the licensing situation for emacs packages provided through Emacs package archives has been under focus. I have volunteered to contact the authors of packages that have a license that is incompatible with Emacs, which is now under GPL-3+.

See https://lists.gnu.org/archive/html/emacs-devel/2017-07/msg01069.html

So I wonder if you could consider to change the license of your package to GPL-3+?

Looking at the files there seems to be a mix of files with different license. The top package is under GPL-2 but there are files and sub modules that are distributed under GPL-3+.

### PR DESCRIPTION
**Pull request by monnier**
This removes all the old code and changes the license of the remaining files to GPLv3+.

Close #198

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,11 +137,11 @@ Supported proof assistants:
 Proof General used to support other proof assistants, but those
 instances are no longer maintained nor available in the MELPA package:
 
-* Legacy support of
-  [Isabelle](https://www.cl.cam.ac.uk/research/hvg/Isabelle/) and
+* Experimental support of: Shell
+* Obsolete instances: Demoisa
+* Removed instances: Twelf, CCC, Hol-Light, ACL2, Plastic, Lambda-Clam, HOL98
   [LEGO](http://www.dcs.ed.ac.uk/home/lego)
```
Comment (erikmd) [inline README.md:143]: Thanks a lot @monnier!

Nitpick in the README.md: one could add two commas here,
```suggestion
* Removed instances: Twelf, CCC, Hol-Light, ACL2, Plastic, Lambda-Clam, HOL98,
  [LEGO](http://www.dcs.ed.ac.uk/home/lego),
```

**Comment by erikmd** [comment]
BTW I guess [the last section of the README.md](https://github.com/ProofGeneral/PG/#contributing) could be rephrased to something like:

> Contributions to this repository are placed under the BSD-3 license. As BSD-3 is compatible with GPLv3+, this means that we can merge them using the same license as the rest of the codebase, while you keep all the rights on your code. For more info, see <https://opensource.org/licenses/BSD-3-Clause> .

File: proof-general.el [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 ;; This software is free software; you can redistribute it and/or
 ;; modify it under the terms of the GNU General Public License
-;; version 2, as published by the Free Software Foundation.
+;; version 3, as published by the Free Software Foundation.
```
Comment (erikmd) [inline proof-general.el:19]: I skimmed the PR diff and it seems the only missing words are these ones:

```suggestion
;; version 3 or later, as published by the Free Software Foundation.
```

But maybe you could just rephrase a bit the whole paragraph to have the very same text as other files?

```
;; This is free software; you can redistribute it and/or modify
;; it under the terms of the GNU General Public License as published by
;; the Free Software Foundation; either version 3, or (at your option)
;; any later version.
```

**Comment by monnier** [comment]
> BTW I guess [the last section of the
> README.md](https://github.com/ProofGeneral/PG/#contributing) could be
> rephrased to something like:
>
>> Contributions to this repository are placed under the BSD-3 license. As
> BSD-3 is compatible with GPLv3+, this means that we can merge them using the
> same license as the rest of the codebase, while you keep all the rights on
> your code. For more info, see <https://opensource.org/licenses/BSD-3-Clause>
> .

Hmm... no matter under which license the contributors send us their
code, they always keep all the rights to their code, so that doesn't
sound right.

Maybe we can just remove this section altogether since we don't have
this dual-license situation any more and can simply accept code under
the GPLv3+ license (or anything sufficiently permissive to be compatible
with it, of course, e.g. BSD3 but not GPLv2).

**Comment by monnier** [comment]
> Nitpick in the README.md: one could add two commas here,
> ```suggestion
> * Removed instances: Twelf, CCC, Hol-Light, ACL2, Plastic, Lambda-Clam, HOL98,
>   [LEGO](http://www.dcs.ed.ac.uk/home/lego),
> ```

Good point, will do.

**Comment by erikmd** [comment]
@monnier

> BTW I guess [the last section of the README.md](https://github.com/ProofGeneral/PG/#contributing) could be rephrased to something like:
> > Contributions to this repository are placed under the BSD-3 license. As BSD-3 is compatible with GPLv3+, this means that we can merge them using the same license as the rest of the codebase, while you keep all the rights on your code. For more info, see <https://opensource.org/licenses/BSD-3-Clause> .
> 
> Hmm... no matter under which license the contributors send us their code, they always keep all the rights to their code,

Sure; actually the underlying reasoning related to this claim was just that (1) we don't require users to transfer their copyright (unlike for example, what is asked by the FSF for direct submissions to emacs), and (2) getting contributions with a compatible permissive license such as BSD-3 would still give the ability to relicense PG to another compatible license (LGPL or GPLv4 or…) if ever this were deemed necessary for the project or asked by the emacs community…

> Maybe we can just remove this section altogether since we don't have this dual-license situation any more and can simply accept code under the GPLv3+ license (or anything sufficiently permissive to be compatible with it, of course, e.g. BSD3 but not GPLv2).

I agree that the dual-license issue is now resolved, and thereby this BSD-3 requirement is not necessary anymore.
(The only potential usefulness being what I recapped above.)

So I don't object to remove this section henceforth to simplify things, and also accept contributions under GPLv3+.
But I'd like to also have a green light (or just a :+1: reaction to my comment :) from our fellow co-maintainers to do so, before we merge your PR − Cc @cpitclaudel @hendriktews @Matafou

**Comment by Matafou** [comment]
You have my green light.
P

**Comment by erikmd** [comment]
Thanks @monnier for your force-push;
when we are on it, actually the autodetected pattern for closing an issue at merge time is `Fix #198` or `Close #198` anywhere in the commit message (not `fixes issue #198`)
and actually, it's enough to have this clause in the *description* of the issue (i.e., your [original post](https://github.com/ProofGeneral/PG/pull/627#issue-1059026800) which I edited it recently so it's OK) **or** in any commit message of the PR as you did.

**Comment by erikmd** [comment]
So Clément and Pierre were fine with my suggestion − to get rid of the "contributing requirement under BSD-3 license" that used to be mentioned in our README, and just stick to GPLv3+.

@hendriktews, OK for you as well?

**Comment by erikmd** [comment]
Meanwhile: feel free @monnier to force-push again to tweak the `Fix #198` keyword in your last commit :)

**Comment by hendriktews** [comment]
> @hendriktews, OK for you as well?

Yes, OK.

**Comment by erikmd** [comment]
OK! thank you all for your feedback ✨ so let's merge #627 :)

**Comment by erikmd** [comment]
So just FYI, GitHub's automatic detection of the license if fine:

2021-11-25_16-09-33_Screenshot_License_OK

I'll check everything is OK from MELPA later on (after the rolling-release deployment of master)

**Comment by cpitclaudel** [comment]
Woohoo, excellent job everyone, and thanks so much @monnier .

**Pull request merged**